src/components/Common/SearchBox.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState, forwardRef } from 'react';
+import remcalc from 'remcalc';
+import styled, { keyframes } from 'styled-components';
+import is from 'styled-is';
+
+import { Input } from './Forms';
+import Anchor from './Anchor';
+
+const fadeIn = keyframes`
+  from{
+    opacity: 0;
+  }
+  to {
+    opacity: 1;
+  }
+`;
+
+export const SearchField = styled.section`
+  position: relative;
+  margin-bottom: ${remcalc(36)};
+`;
+
+export const Results = styled.ul`
+  width: 100%;
+  max-height: ${remcalc(200)};
+  list-style-type: none;
+  background: ${props => props.theme.colors.greyBg};
+  position: absolute;
+  left: 0;
+  top: ${remcalc(56)};
+  z-index: 10;
+  overflow-x: auto;
+  overflow-y: scroll;
+  will-change: opacity;
+  animation: ${fadeIn} ${props => props.theme.animations.long} linear;
+`;
+
+export const Result = styled.li`
+  padding: ${remcalc(8)} ${remcalc(24)};
+  font-size: ${remcalc(18)};
+  line-height: ${remcalc(24)};
+  font-weight: 300;
+  color: ${props => props.theme.colors.text};
+  background: transparent;
+  transition: background ${props => props.theme.animations.normal} ease-in-out,
+    color ${props => props.theme.animations.normal} ease-in-out;
+  &:hover {
+    background: ${props => props.theme.colors.vibrant};
+    > a {
+      color: ${props => props.theme.colors.blueBg};
+    }
+  }
+  ${is('notFound')`
+    pointer-events:none;
+    color: ${props => props.theme.colors.grey};
+    font-weight: 500;
+  `};
+`;
+
+export const SearchBox = forwardRef(
+  ({ placeholder = 'Search...', searchedData }, ref) => {
+    const [isListBoxOpen, setOpen] = useState(false);
+    const [results, setResults] = useState([]);
+
+    const handleChange = () => {
+      const newResults = searchedData();
+      if (ref.current.value.length >= 2) {
+        setResults(newResults);
+        setOpen(true);
+      } else {
+        setOpen(false);
+      }
+    };
+
+    return (
+      <SearchField>
+        <Input
+          ref={ref}
+          type="search"
+          placeholder={placeholder}
+          aria-label="Search"
+          onChange={handleChange}
+          noBoxShadow
+          searchBox
```

Comment:
This renders inconsistently on different browsers. Some thing I've noticed:

1. Input padding and border-radius ignored on Safari, causing a split distance between the input and the result and outline to be off
![Screenshot 2020-06-16 at 11 07 06](https://user-images.githubusercontent.com/25818826/84761594-8a2d6080-afc1-11ea-82db-084937928061.png)

2. We're letting the browser decide the input clear button styling: a blue cross that appears when hovered on Chrome; a cross inside a grey circle always visible when the input has some value on Safari; Inexistent on Firefox. For consistency sake we should probably hide it for all browsers IMO
![Screenshot 2020-06-16 at 11 11 28](https://user-images.githubusercontent.com/25818826/84762050-38390a80-afc2-11ea-9f15-2326cbb8128a.png)

This is probably caused by having _searchbox_ as type. I guess we need some style overrides on the browser defaults.

Reply:
looking good now with type _text_ 👍